cerberus/base.py

```diff
@@ -1439,9 +1439,9 @@ def __validate_definitions(self, definitions, field):
     def _validate_allowed(self, allowed_values, field, value):
         """ {'type': 'container_but_not_string'} """
         if isinstance(value, Iterable) and not isinstance(value, str):
-            unallowed = set(value) - set(allowed_values)
+            unallowed = [v for v in value if v not in allowed_values]
```

Comment:
i'd prefer:

- to store a tuple as error information with `tuple(<comprehension expression>)`
- use `x` inside the comprehension as `v` is commonly used when iterating over a mapping's values